### PR TITLE
refactor: replace all the datavolumes and pvcs name

### LINKED ISSUE #11254: refactor: replace all the datavolumes and pvcs name assignements  with PVCNameFromVirtVolume
**Is your feature request related to a problem? Please describe**:
We have in multiple places with code like in [this case](https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-handler/vm.go#L2477-L2482) that can be refactored with the function [PVCNameFromVirtVolume](https://github.com/kubevirt/kubevirt/blob/main/pkg/storage/types/pvc.go#L98C6-L98C27)

**Describe the solution you'd like**:
Find all the places and replace the code with the function
This would be a good issue for people willing to get more familiar with the code for GSoC

### PR DESCRIPTION
**Pull request by sarthaksarthak9**
replace all the datavolumes and pvcs name assignements with PVCNameFromVirtVolume

### What this PR does

Fixes #11254 

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Special notes for your reviewer
Thank you for reviewing

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by sarthaksarthak9** [comment]
@victortoso could you pls review the pr !!

**Comment by alicefr** [comment]
@sarthaksarthak9 thanks for your work, the changes look fine to me

**Comment by alicefr** [comment]
@sarthaksarthak9 please add `NONE` to the release note

File: pkg/storage/export/export/export.go [hunk 1]
```diff
@@ -1342,13 +1342,7 @@ func (ctrl *VMExportController) expandVirtualMachine(vm *virtv1.VirtualMachine)
...
 func (ctrl *VMExportController) updateHttpSourceDataVolumeTemplate(vm *virtv1.VirtualMachine) *virtv1.VirtualMachine {
 	for _, volume := range vm.Spec.Template.Spec.Volumes {
-		volumeName := ""
-		if volume.DataVolume != nil {
-			volumeName = volume.DataVolume.Name
-		}
-		if volume.PersistentVolumeClaim != nil {
-			volumeName = volume.PersistentVolumeClaim.ClaimName
-		}
+		volumeName := types.PVCNameFromVirtVolume(&volume)
```
Comment (alicefr) [inline pkg/storage/export/export/export.go:1345]: You can inline this with the if statement

Comment (alicefr) [reply pkg/storage/export/export/export.go:1345]: I mean:
```suggestion
		if volumeName := types.PVCNameFromVirtVolume(&volume); volumeName != "" {
			vm.Spec.DataVolumeTemplates = ctrl.replaceUrlDVTemplate(volumeName, vm.Namespace, vm.Spec.DataVolumeTemplates)
		}
```

**Comment by alicefr** [comment]
@sarthaksarthak9 please, adjust your commit message. Right now, you have the sign-off as title.
Please, give a short title and then a quick description of what the commit does.
You can add the signoff using `git commit -s`  and you can use `git commit --amend -s` to modify the latest commit.

File: pkg/storage/export/export/export.go [hunk 2]
```diff
@@ -1342,15 +1342,8 @@ func (ctrl *VMExportController) expandVirtualMachine(vm *virtv1.VirtualMachine)
...
-		volumeName := ""
-		if volume.DataVolume != nil {
-			volumeName = volume.DataVolume.Name
-		}
-		if volume.PersistentVolumeClaim != nil {
-			volumeName = volume.PersistentVolumeClaim.ClaimName
-		}
-		if volumeName != "" {
-			vm.Spec.DataVolumeTemplates = ctrl.replaceUrlDVTemplate(volumeName, vm.Namespace, vm.Spec.DataVolumeTemplates)
+		if types.PVCNameFromVirtVolume(&volume) != "" {
```
Comment (alicefr) [inline pkg/storage/export/export/export.go:1345]: Please, use here a variable for the name. Conceptually, this isn't wrong, but the variable avoids to call the function twice

**Comment by alicefr** [comment]
@sarthaksarthak9 can you please fix the commit message with a short title and a brief explanation why the change is needed

**Comment by alicefr** [comment]
@sarthaksarthak9 please, give a short title and a message body in the commit message. Try to give a look to the git history with `git log` to standardize your commit message.

**Comment by victortoso** [comment]
@sarthaksarthak9 

Please take a look at this: https://cbea.ms/git-commit/
The first line is usually a short log, to provide quick context. The body of the commit log provides further context and material needed. This is super helpful when you have to look at hundreds of commits or even better understands the changes thanks to the author's context.

Thanks for working on this.

**Review by xpivarc — COMMENTED** [review]
/approve
/hold
@victortoso  PTAL

**Comment by victortoso** [comment]
>  @victortoso PTAL
/lgtm (too)

**Comment by avlitman** [comment]
Hi this pr is waiting for long time. 
@xpivarc can we remove the hold?
@sarthaksarthak9 can you rebase the pr?

**Comment by avlitman** [comment]
Hi @xpivarc can we cancel the hold?

**Comment by avlitman** [comment]
@xpivarc any chance you can take a look why ci is failing?

**Comment by enp0s3** [comment]
/hold
@sarthaksarthak9 You have a compilation error:
```
11:02:39: pkg/storage/export/export/export.go:1365:86: too many arguments in call to ctrl.replaceUrlDVTemplate
11:02:39: 	have (string, string, []"kubevirt.io/api/core/v1".DataVolumeTemplateSpec)
11:02:39: 	want (string, []"kubevirt.io/api/core/v1".DataVolumeTemplateSpec)
```

**Comment by enp0s3** [comment]
@sarthaksarthak9 You can test your changes locally by running `make test`, thus ensuring everything is built properly

**Pull request closed without merging**